cpp/include/cudf/digitize.hpp

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/column/column_view.hpp>
+#include <cudf/column/column.hpp>
+#include <cudf/types.hpp>
+#include <memory>
+
+namespace cudf {
+
+/**
+ * @brief Finds the indices of the bins in which each value of the column
+ * belongs.
+ *
+ * For `x` in `col`, if `right == false` this function finds
+ * `i` such that `bins[i-1] <= x < bins[i]`. If `right == true`, it will find `i`
+ * such that `bins[i - 1] < x <= bins[i]`. Finally, if `x < bins[0]` or
+ * `x > bins[num_bins - 1]`, it sets the index to `0` or `num_bins`, respectively.
+ *
+ * NOTE: This function does not handle null values and will throw if `col`
+ * or `bins` contain any.
+ * 
+ * @throws cudf::logic_error if `col` and `bins` types mismatch
+ * @throws cudf::logic_error if type is not numeric or timestamp
```

Comment:
If you always use the `element_relational_comparator`, this will also work for strings.

Reply:
Done